File: code-examples/java-rs/authentication/sha1.java
import com.mongodb.ConnectionString;
import com.mongodb.reactivestreams.client.MongoClients;
import com.mongodb.reactivestreams.client.MongoClient;
import org.bson.Document;
import java.util.Arrays;

public class Main {
    private MongoClient mongoClient;

    public static void main(String[] args) {
        new Main().connectToDatabase();
    }

    private void connectToDatabase() {
        char[] password = "<password>".toCharArray(); 
        String username = "<username>";
        String clusterUrl = "<MongoDB cluster url>";

        ConnectionString connectionString = new ConnectionString(
                "mongodb+srv://" + username + ":" + new String(password) + "@" + clusterUrl + "/?authSource=admin&authMechanism=SCRAM-SHA-1"
        );

        mongoClient = MongoClients.create(connectionString);

        mongoClient.getDatabase("admin").runCommand(new Document("ping", 1))
                .subscribe(success -> System.out.println("Connected successfully to server"),
                        error -> System.out.println("An error occurred: " + error.getMessage()),
                        this::disconnectFromDatabase);
    }

    private void disconnectFromDatabase() {
        mongoClient.close();
    }
}

File: code-examples/scala/authentication/sha1.scala
import org.mongodb.scala.{MongoClient, Document}
import scala.concurrent.Await
import scala.concurrent.duration._
import scala.util.{Failure, Success}

// Enter your MongoDB cluster URL, username, and password
val username = "<username>"
val password = "<password>"
val clusterUrl = "<MongoDB cluster url>"

val authMechanism = "SCRAM-SHA-1"

val mongoClient: MongoClient = MongoClient(s"mongodb+srv://$username:$password@$clusterUrl/?authMechanism=$authMechanism")

try {
  val database = mongoClient.getDatabase("admin")
  val command = Document("ping" -> 1)
  val result = database.runCommand(command).head()

  Await.result(result, 10.seconds) // Waiting for the results from MongoDB server
  println("Connected successfully to server")
} catch {
  case t: Throwable => println("An error has occurred: " + t.getMessage)
} finally {
  mongoClient.close()
}
